File: backend/src/main/java/com/devops/dashboard/service/TaskService.java
package com.devops.dashboard.service;

import com.devops.dashboard.dto.common.PagedResponse;
import com.devops.dashboard.dto.task.CommentRequest;
import com.devops.dashboard.dto.task.TaskCommentResponse;
import com.devops.dashboard.dto.task.TaskRequest;
import com.devops.dashboard.dto.task.TaskResponse;
import com.devops.dashboard.dto.task.TaskStatusUpdateRequest;
import org.springframework.web.multipart.MultipartFile;

public interface TaskService {
    PagedResponse<TaskResponse> getTasks(String search, String status, Long projectId, Long assignedUserId, int page, int size, String currentUserEmail);
    TaskResponse createTask(TaskRequest request, String actorEmail);
    TaskResponse updateTask(Long id, TaskRequest request, String actorEmail);
    void deleteTask(Long id, String actorEmail);
    TaskResponse updateStatus(Long id, TaskStatusUpdateRequest request, String actorEmail);
    TaskCommentResponse addComment(Long taskId, CommentRequest request, String actorEmail);
    TaskResponse uploadAttachment(Long taskId, MultipartFile file, String actorEmail);
}

File: frontend/src/services/taskService.js
import { api } from './api';

export const taskService = {
  getTasks: async (params) => {
    const response = await api.get('/tasks', { params });
    return response.data;
  },
  createTask: async (payload) => {
    const response = await api.post('/tasks', payload);
    return response.data;
  },
  updateTask: async (id, payload) => {
    const response = await api.put(`/tasks/${id}`, payload);
    return response.data;
  },
  deleteTask: async (id) => {
    const response = await api.delete(`/tasks/${id}`);
    return response.data;
  },
  updateStatus: async (id, payload) => {
    const response = await api.put(`/tasks/${id}/status`, payload);
    return response.data;
  },
  addComment: async (id, payload) => {
    const response = await api.post(`/tasks/${id}/comments`, payload);
    return response.data;
  },
  uploadAttachment: async (id, file) => {
    const formData = new FormData();
    formData.append('file', file);
    const response = await api.post(`/tasks/${id}/attachments`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return response.data;
  },
};
